packages/fuel-indexer/src/commands/run.rs

```diff
@@ -57,56 +100,82 @@ pub async fn exec(args: IndexerArgs) -> anyhow::Result<()> {
 
     let service_handle = tokio::spawn(service.run());
 
+    // for graceful shutdown
```

Comment:
@lostman 

- Just a heads up here.
- I see the logic of how the service starts and is feature flagged has been changed
- Totally fine with that overall, but just an FYI that the manual testing to verify this works is gonna be a bit vigorous
- You will need to test:
  - The service starts with all/any features `--all-features`, `--no-default-features`
  - The service starts with `--local-fuel-node`
  - Native execution starts with all/any features `--all-features`, `--no-default-features`
  - Naive execution starts with `--local-fuel-node`
  - Docker builds still work
    - Hello-indexer example
    - Block-explorer example
    - Hello-indexer-native example
- And we'd need to update the PR testing steps to reflect this 
  - You can see something somewhat similar over in https://github.com/FuelLabs/fuel-indexer/pull/772